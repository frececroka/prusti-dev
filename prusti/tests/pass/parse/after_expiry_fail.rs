// compile-flags: -Zprint-desugared-specs -Zprint-typeckd-specs -Zskip-verify -Zhide-uuids
// normalize-stdout-test: "[a-z0-9]{32}" -> "$(NUM_UUID)"
// normalize-stdout-test: "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}" -> "$(UUID)"

#![feature(register_tool)]
#![register_tool(prusti)]

use prusti_contracts::*;

#[after_expiry_if(a)]
fn test1() {}

#[after_expiry_if(a => a)]
fn test2() {}

#[pledge(a, a)]
fn test3() {}

fn main() {}
