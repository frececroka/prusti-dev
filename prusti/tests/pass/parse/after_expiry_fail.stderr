error: expected `,`
  --> $DIR/after_expiry_fail.rs:10:19
   |
10 | #[after_expiry_if(a)]
   |                   ^

error: expected `,`
  --> $DIR/after_expiry_fail.rs:13:24
   |
13 | #[after_expiry_if(a => a)]
   |                        ^

error: unexpected token
  --> $DIR/after_expiry_fail.rs:16:17
   |
16 | #[pledge(a, a)]
   |           ^

error: aborting due to 3 previous errors

